### ISSUE TITLE
rainbow variable does not take effect after IDEA is restarted

### ISSUE THREAD

**Issue by fengzengfly** [izhangzhihao/intellij-rainbow-brackets#2551]
Your issue may already be reported!
Please search on the [issues](https://github.com/izhangzhihao/intellij-rainbow-brackets/issues) and the [document](https://github.com/izhangzhihao/intellij-rainbow-brackets/blob/2020.3/README.md) before creating one.

## Please check

- [√ ] I already support this project.
- [√] If you are submitting a feature request, please do consider buy this plugin [here](https://www.jetbrains.com/shop/buy?item=P:N:PRAINBOWBRACKET:M) or donating us on [Open Collective](https://opencollective.com/intellij-rainbow-brackets) Or by AliPay/WeChatPay.
- [ √] This issue/feature request is not reported before.

## Your programming languages
java
## Expected Behavior
Rainbow variable is always in effect

## Current Behavior
After restarting the idea, the rainbow variable disappears, but the rainbow bracket takes effect

## Possible Solution
Not obligatory, but suggest a fix/reason for the bug, or ideas how to implement the addition or change.
Or what have you tried to resolve this issue.

## Code snippet for reproduce (for bugs)
![image](https://user-images.githubusercontent.com/48151231/201152437-760fd7c8-0455-496e-8145-78c4ffbe32e4.png)

## Your Environment

* Plugin version:
* IDE & Operating System version, comment your env as below(go to "About IntelliJ IDEA" -> click the "copy" icon):
* NOTE: If you are going to report a bug but WITHOUT your env information, your issue might be closed directly.

```
IntelliJ IDEA 2022.2.3 (Ultimate Edition)
Build #IU-222.4345.14, built on October 5, 2022
Licensed to 冯 曾
Subscription is active until November 4, 2023.
For educational use only.
Runtime version: 17.0.4.1+7-b469.62 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 20480M
Cores: 16
Non-Bundled Plugins:
    leetcode-editor (8.4)
    com.tabnine.TabNine (0.7.27)
    com.nasller.CodeGlancePro (1.5.3)
    org.jetbrains.kotlin (222-1.7.21-release-272-IJ4167.29)
    com.intellij.guice (222.4345.14)
    com.github.setial (4.0.2)
    com.github.chencn.yamlandprops (1.0.5)
    com.alibaba.p3c.smartfox (2.1.1)
    Statistic (4.2.6)
    Key Promoter X (2022.3)
    GrepConsole (12.15.211.6693.1)
    zielu.gittoolbox (212.9.9)
    mobi.hsz.idea.gitignore (4.4.2)
    MavenRunHelper (4.23.222.2964.0)
    JRebelPlugin (2022.4.1)
    izhangzhihao.rainbow.brackets (2022.3.3.1-ij)
    com.baomidou.plugin.idea.mybatisx (1.5.5)
    cn.yiiguxing.plugin.translate (3.3.5)
    jclasslib (6.0.4.1)

Kotlin: 222-1.7.21-release-272-IJ4167.29
```

**Comment by hwooo** [comment]
Same here. I have to turn off and on again "Enable rainbow variables" option after restart IDEA.

**Comment by izhangzhihao** [comment]
[intellij-rainbow-brackets-2022.3.4-ij.zip](https://github.com/izhangzhihao/intellij-rainbow-brackets/files/9985598/intellij-rainbow-brackets-2022.3.4-ij.zip)

Hi, please use this for now.

**Comment by fengzengfly** [comment]
> [intellij-rainbow-brackets-2022.3.4-ij.zip](https://github.com/izhangzhihao/intellij-rainbow-brackets/files/9985705/intellij-rainbow-brackets-2022.3.4-ij.zip)
> 
> Hi, please use this for now.

This problem still exists，my  environment:
```
IntelliJ IDEA 2022.2.3 (Ultimate Edition)
Build #IU-222.4345.14, built on October 5, 2022
Licensed to xx
Subscription is active until November 4, 2023.
For educational use only.
Runtime version: 17.0.4.1+7-b469.62 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 20480M
Cores: 16
Non-Bundled Plugins:
    leetcode-editor (8.4)
    com.tabnine.TabNine (0.7.27)
    com.nasller.CodeGlancePro (1.5.3)
    org.jetbrains.kotlin (222-1.7.21-release-272-IJ4167.29)
    com.intellij.guice (222.4345.14)
    com.github.setial (4.0.2)
    com.github.chencn.yamlandprops (1.0.5)
    com.alibaba.p3c.smartfox (2.1.1)
    Statistic (4.2.6)
    Key Promoter X (2022.3)
    GrepConsole (12.15.211.6693.1)
    zielu.gittoolbox (212.9.9)
    mobi.hsz.idea.gitignore (4.4.2)
    MavenRunHelper (4.23.222.2964.0)
    JRebelPlugin (2022.4.1)
    izhangzhihao.rainbow.brackets (2022.3.4-ij)
    com.baomidou.plugin.idea.mybatisx (1.5.5)
    cn.yiiguxing.plugin.translate (3.3.5)
    jclasslib (6.0.4.1)

Kotlin: 222-1.7.21-release-272-IJ4167.29
```

**Comment by izhangzhihao** [comment]
Can you check the settings here:

<img width="499" alt="image" src="https://user-images.githubusercontent.com/12044174/201249309-4588328e-0b13-458a-b02a-525bf56b52ef.png">

It should work (I edit the uploaded file once, could you make sure the new files works or not?)

**Comment by izhangzhihao** [comment]
I test here, it works after restart.

```
IntelliJ IDEA 2022.2.3 (Ultimate Edition)
Build #IU-222.4345.14, built on October 5, 2022
Licensed to IntelliJ Rainbow Brackets / Zhang Zhihao
Subscription is active until January 20, 2023.
For non-commercial open source development only.
Runtime version: 17.0.4.1+7-b469.62 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.6.1
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 16
Metal Rendering is ON
Registry:
    debugger.new.tool.window.layout=true
    ide.plugins.snapshot.on.unload.fail=true
    ide.tooltip.initialDelay=900
    ide.experimental.ui=true
    ide.balloon.shadow.size=15
    scala.erase.compiler.process.jdk.once=false

Non-Bundled Plugins:
    org.nik.presentation-assistant (1.0.10)
    net.sf.opk.avro-schema-support (213.2.1)
    net.seesharpsoft.intellij.plugins.csv (2.19.0)
    com.sdadas.jasypt (0.9.3)
    com.mallowigi.idea (18.1.0)
    org.jetbrains.kotlin (222-1.7.20-release-201-IJ4167.29)
    com.github.copilot (1.1.33.1980)
    atom-one (1.5.1)
    String Manipulation (9.5.1)
    org.sonarlint.idea (7.1.1.54565)
    io.github.isliudong.easyRequest (1.0.1-SNAPSHOT)
    com.github.beansoft.spring.mybatis (2020.2.0 EAP)
    com.intellij.queryComplexity (222.4345.14)
    org.utbot.intellij.plugin.id (2022.10)
    io.github.newhoo.restkit (2.1.1)
    org.intellij.scala (2022.2.13)
    com.intellij.bigdatatools (222.3345.33)
    com.alibabacloud.intellij.toolkit-intellij (2022.9.1-222)
    izhangzhihao.rainbow.brackets (2022.3.4-ij)

Kotlin: 222-1.7.20-release-201-IJ4167.29
```

**Comment by fengzengfly** [comment]
> Can you check the settings here:
> 
> <img alt="image" width="499" src="https://user-images.githubusercontent.com/12044174/201249309-4588328e-0b13-458a-b02a-525bf56b52ef.png">
> 
> It should work (I edit the uploaded file once, could you make sure the new files works or not?)

yes，i'm sure I turned on these settings，but it still doesn't work.
![image](https://user-images.githubusercontent.com/48151231/201249977-b6003c1b-9cce-464d-a711-93458abd0b06.png)

**Comment by izhangzhihao** [comment]
Try this one: [intellij-rainbow-brackets-2022.3.4-ij.zip](https://github.com/izhangzhihao/intellij-rainbow-brackets/files/9985705/intellij-rainbow-brackets-2022.3.4-ij.zip)

**Comment by izhangzhihao** [comment]
Is this setting enabled automaticly?

<img width="938" alt="image" src="https://user-images.githubusercontent.com/12044174/201250718-d1968edc-f778-485b-af82-a865db2c78c2.png">

**Comment by fengzengfly** [comment]
> Try this one
> 
> Did you restart IDE after install the plugin?

yes，i restart IDE after install the plugin。

**Comment by izhangzhihao** [comment]
Maybe you just need to edit the files so that IDE will re-highlighting the editor again.

**Comment by fengzengfly** [comment]
> Is this setting enabled automaticly?
> 
> <img alt="image" width="938" src="https://user-images.githubusercontent.com/12044174/201250718-d1968edc-f778-485b-af82-a865db2c78c2.png">

thank you，  it can work after opening this setting

**Comment by izhangzhihao** [comment]
this setting should be checked if you enable "Rainbow Brackets -> enable rainbow variables", please restart your IDE to verify that.

**Comment by izhangzhihao** [comment]
Should fixed in 2022.3.4
